### ISSUE TITLE
no-invalid-this warns when used in class arrow functions

### ISSUE THREAD

**Issue by mikew** [eslint/eslint#7565]
**Tell us about your environment**

* **ESLint Version:**
v3.9.1
* **Node Version:**
v6.9.1
* **npm Version:**
3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

```js
module.exports = {
  rules: {
    'no-invalid-this': 'warn'
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
class Test {
  someFunction = () => {
    this.foo()
  }
}
```

**What did you expect to happen?**
No complaints about `no-invalid-this`.

**What actually happened? Please include the actual, raw output from ESLint.**
```
Unexpected 'this' (no-invalid-this)
```

**Comment by not-an-aardvark** [comment]
Hi,

I think this is happening because class properties are an experimental feature, and ESLint's core rules generally don't support experimental features.

You might want to file an issue on [eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel) instead.

**Comment by not-an-aardvark** [comment]
I'm going to close this issue since it's not a problem with ESLint, but please feel free to continue the discussion.

**Comment by mikew** [comment]
Thanks for pointing in the right direction. If anyone gets bit by this, it seems it's already fixed in [`eslint-plugin-babel`](https://github.com/babel/eslint-plugin-babel/blame/026eab82b887d70b5f8238d1b33a3284924bf378/README.md#L54)
